docs/DeveloperGuide.md

```diff
@@ -1,38 +1,337 @@
+
+The `Module` class serves as a wrapper around a `TaskList`, providing additional attributes including the module code and module description. Within the context of Mod Happy, modules can be viewed as task categories with names, descriptions and other attributes; for this reason, the General Tasks list is implemented as a `Module` under the hood.
+
+> 📔 <span style="color:#00bb00">**NOTE:**</span>
+>
+> An alternative method of implementing `ModuleList` is shown below, where the default General Tasks
+list is simply represented as a `TaskList` instead of a full-fledged `Module`.
+>
+> ![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/DataAlternative.puml)
+>
+> While this model is arguably closer to real life, the program logic would have to operate on different object types depending on whether a given `Task` belongs to a user-created Module or the default General Tasks list. This was deemed to increase coupling and introduce too much unnecessary clutter to the code, hence it was not used.
+
+<br>
+
+### 5.4. Command Component
+
+The `Command` component is in charge of actually executing the operations requested by the user.
+
+The following partial class diagram illustrates the structure of this component:
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/CommandClassDiagram.puml)
+
+All commands inherit the abstract `Command` class and must contain an `execute()` method. The program logic that must be executed to fulfil the requested command is implemented in this method. Additionally, `execute()` returns any command output to be displayed to the user as feedback.
+
+`Command` instances are generated by their corresponding `Parser` classes (e.g. `AddCommand` is constructed by `AddParser`) and executed by `Main`.
+
+<br>
+
+### 5.5. Storage Component
+
+The `Storage` component is responsible for the saving and loading of program data from and to its data files. The following class diagram illustrates the structure of this component:
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/Storage.puml)
+
+Storage is an interface supporting write/read data to/from computer storage:
+* Storage interface is implemented by JsonStorage in Mod Happy, which will read and load data to and from json format.
+* ListStorage can save a ArrayList of any class that extends Object in json format, and read them back into corresponding objects. (E.g. ModuleListStorage, TaskListStorage inherit from ListStorage)
+* There are navigability to Storage from Main and SaveCommand, which handles the load and write data to/from disk respectively.
+  <br><br><br>
+
+## 6. Implementation
+
+This section describes some details on how some features are implemented.
```

Comment:
A bit redundant given that the title of the section, "Implementation," already implies this - perhaps you could provide more detailed explanation, like "This section provides additional detail on how some core features, such as _____, are implemented."